ga4gh/datamodel/datasets.py

```diff
@@ -190,3 +237,70 @@ def _setMetadata(self):
                 except KeyError as err:
                     raise exceptions.MissingDatasetMetadataException(
                         metadataFileName, str(err))
+
+
+class AbstractOntology:
+    """
+    The base class of storing an ontology
+    """
+
+    def __init__(self):
+        self.nameById = dict()
+        self.idByName = dict()
+
+    def add(self, id_, name):
+        self.nameById[id_] = name
+        self.idByName[name] = id_
+
+    def getId(self, name):
+        return self.idByName[name]
+
+    def getName(self, id_):
+        return self.nameById[id_]
+
+
+class AbstractOntologies(object):
+    """
+    The base class of storing ontologies
+    """
+
+    def __init__(self):
+        self.ontologyByName = dict()
+
+    def add(self, ontologyName, ontology):
+        self.ontologyByName[ontologyName] = ontology
+
+    def get(self, ontologyName):
+        return self.ontologyByName[ontologyName]
+
+    def len(self):
+        return len(self.ontologyByName)
+
+
+class FileSystemOntology(AbstractOntologies):
+    """
+    An ontology read from the file system.
+    Simple implementation using tab separated TXT file: "id\tname"
+    """
+    def __init__(self, localId, dataDir, backend):
+        super(FileSystemOntology, self).__init__()
+        self.readOntologies(dataDir)
+
+    def readOntologies(self, dataDir):
+        self._dataDir = dataDir
+        # Find TXT files
+        for filename in os.listdir(dataDir):
+            if fnmatch.fnmatch(filename, '*.txt'):
+                ontologyName, _ = os.path.splitext(filename)
+                path = os.path.join(dataDir, filename)
+                ontology = self.readOntology(path)
+                self.add(ontologyName, ontology)
+
+    def readOntology(self, filename):
+        ontology = AbstractOntology()
```

Comment:
OK, I see what's happening now. I'm confused about the plurality here though. Is one 'ontology' a single line in the text file, or is the the whole file? If one line is an ontology, then we should consider renaming the class to OntologySet, and trying to follow the design for the other XSet objects. If not, we should use OntologyTerm a bit more prominently and be clear that an ontology maps to set of ontology terms.